astropy/time/formats.py

```diff
@@ -153,11 +153,11 @@ def _check_val_type(self, val1, val2):
                             .format(self.name))
 
         if getattr(val1, 'unit', None) is not None:
-            # set possibly scaled unit any quantities should be converted to
+            # Possibly scaled unit any quantity-likes should be converted to
             _unit = u.CompositeUnit(getattr(self, 'unit', 1.), [u.day], [1])
-            val1 = val1.to(_unit).value
+            val1 = u.Quantity(val1, copy=False).to_value(_unit)
```

Comment:
At first, I just changed to `to_value`, but this failed, since one of the tests uses `Column` (which has a `to` method, but not a `to_value`). When I noticed that, I felt it was actually better not to assume having a `unit` goes with having `to` method either, but rather just let the `Quantity` initializer deal with different objects that have a `.unit` attribute.